src/components/application_manager/include/application_manager/application.h

```diff
@@ -786,6 +786,60 @@ class Application : public virtual InitialApplicationData,
    */
   virtual uint32_t GetAvailableDiskSpace() = 0;
 
+#ifdef SDL_REMOTE_CONTROL
+  /**
+   * @brief set_system_context Set system context for application
+   * @param system_context Current context
+   */
+  virtual void set_system_context(
+      const mobile_api::SystemContext::eType& system_context) = 0;
+
+  /**
+   * @brief set_audio_streaming_state Set audio streaming state for application
+   * @param state Current audio streaming state
+   */
+  virtual void set_audio_streaming_state(
+      const mobile_api::AudioStreamingState::eType& state) = 0;
+
+  /**
+   * @brief set_hmi_level Set HMI level for application
+   * @param hmi_level Current HMI level
+   */
+  virtual void set_hmi_level(const mobile_api::HMILevel::eType& hmi_level) = 0;
+
+  /**
+   * @brief Return pointer to extension by uid
+   * @param uid uid of extension
+   * @return Pointer to extension, if extension was initialized, otherwise NULL
+   */
+  virtual AppExtensionPtr QueryInterface(AppExtensionUID uid) = 0;
+
+  /**
+   * @brief Add extension to application
+   * @param extension pointer to extension
+   * @return true if success, false if extension already initialized
+   */
+  virtual bool AddExtension(AppExtensionPtr extention) = 0;
```

Comment:
@JackLivio app extention is interface that is able to extend functionality of make SDL application without changing application class and impact to other modules and giving access to other modules.